client/src/client.rs

```diff
@@ -602,7 +610,16 @@ impl<R: Reactor> handle::Handle for Handle<R> {
     }
 
     fn wait_for_height(&self, h: Height) -> Result<BlockHash, handle::Error> {
-        // TODO: Should return immediately if we are already at that height.
+        // Get current block height.
+        let (current_height, _) = self.get_tip()?;
+
+        // Return block hash of height if current height exceeds.
+        if current_height >= h {
+            return self
+                .get_block_by_height(h)?
+                .ok_or(handle::Error::InvalidBlockHeight);
+        }
+
```

Comment:
I think this could be potentially simplified, because if `get_block_by_height` returns `None`, it must mean that `h > current_height`, ie. the height is in the future.

Reply:
Fixed, thanks